presto-main/src/main/java/com/facebook/presto/sql/planner/plan/EnforceUniqueColumns.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class EnforceUniqueColumns
+        extends PlanNode
+{
+    private final PlanNode source;
+    private final Symbol mostSigBitsSymbol;
```

Comment:
8-10 bits for stage I'd should be more than sufficient. It's rare to see a well-behaved query with more than a few dozen stages.

Task ids are per stage, and there's one per machine, so 12-14 bits should cover any reasonable cluster sizes for the foreseeable future.

32 bits for row id are probably not sufficient. We've seen tasks that generate more than that many rows. I'd use more bits for that.